pvapy/cli/adSimServer.py

```diff
@@ -215,18 +308,32 @@ def __init__(self, inputDirectory, inputFile, mmapMode, hdfDataset, hdfCompressi
         self.frameGeneratorList = []
         self.frameCacheSize = max(cacheSize, self.MIN_CACHE_SIZE)
         self.nFrames = nFrames
+        self.config_file = None
 
         inputFiles = []
         if inputDirectory is not None:
             inputFiles = [os.path.join(inputDirectory, f) for f in os.listdir(inputDirectory) if os.path.isfile(os.path.join(inputDirectory, f))]
         if inputFile is not None:
             inputFiles.append(inputFile)
         allowedHdfExtensions = ['h5', 'hdf', 'hdf5']
+        allowedNpExtensions = ['npy', 'npz', 'NPY']
+        if nFrames > 0:
+            inputFiles = inputFiles[:nFrames]
+        if cfgFile is not None and yaml is not None and altFormat > 0:
+            self.config_file = yaml.load(open(cfgFile, 'r'), Loader=yaml.CLoader)
+            if self.config_file['file_info']['ordered_files'] is not None:
+                    inputFiles = self.config_file['file_info']['ordered_files']
         for f in inputFiles:
             ext = f.split('.')[-1]
             if ext in allowedHdfExtensions:
                 self.frameGeneratorList.append(HdfFileGenerator(f, hdfDataset, hdfCompressionMode))
+            elif ext not in allowedNpExtensions and altFormat > 0:
+                fabioFG = FabIOFileGenerator(f, self.config_file)
+                if fabioFG.isLoaded():
+                    self.frameGeneratorList.append(fabioFG)
             else:
+                if ext not in allowedNpExtensions and altFormat <= 0:
```

Comment:
Is the altFormat variable actually needed? If file extension is not hdf5 or numpy, we can simply assume that this is alternate file format, which would simplify interface and this code.

Reply:
I don't think its actually necessary. I didn't want to assume though, I will remove it.

Reply:
Okay,then we do need -aff either.

Reply:
I removed aff entirely.